katcp/testutils.py

```diff
@@ -2049,3 +2051,41 @@ def _advance(self, future):
         yield self.test_instance.set_ioloop_time(
             self.test_instance.ioloop_time + self.quantum)
         future.set_result(None)
+
+
+@contextmanager
+def assert_no_memory_leaks():
+    """Context handler that asserts no memory is leaking within.
```

Comment:
@bmerry PR still in draft, but can you please have a look at this approach to checking for memory leaks?  It seems to pick up the problems.

Suggest first looking at this context handler, and then [its tests](https://github.com/ska-sa/katcp-python/pull/206/files#diff-b7551ec24900e8af18e8858ea2d00622R208).